corehq/form_processor/interfaces/dbaccessors.py

```diff
@@ -396,7 +396,10 @@ def db_accessor(self):
     def get_case(self, case_id):
         if not case_id:
             raise CaseNotFound
-        return self.db_accessor.get_case(case_id)
+        case = self.db_accessor.get_case(case_id)
+        if case.domain != self.domain:
+            raise CaseNotFound
```

Comment:
Can this be change to include the case id so the error message is more informative about which case was not found?
```suggestion
            raise CaseNotFound(case_id)
```

Reply:
Changed!